pennylane/workflow/jacobian_products.py

```diff
@@ -210,6 +210,24 @@ def execute_and_compute_jacobian(self, tapes: QuantumScriptBatch) -> tuple[Resul
         """
 
 
+class NoGradients(JacobianProductCalculator):
+    """A jacobian product calculator that raises errors when a vjp or jvp is requested."""
+
+    def compute_jacobian(self, tapes: QuantumScriptBatch) -> tuple:
+        raise qml.QuantumFunctionError("Derivatives cannot be calculated with diff_method=None")
+
+    def compute_vjp(self, tapes: QuantumScriptBatch, dy: Sequence[Sequence[TensorLike]]) -> tuple:
+        raise qml.QuantumFunctionError("Derivatives cannot be calculated with diff_method=None")
+
+    def execute_and_compute_jvp(
+        self, tapes: QuantumScriptBatch, tangents: Sequence[Sequence[TensorLike]]
+    ) -> tuple[ResultBatch, tuple]:
+        raise qml.QuantumFunctionError("Derivatives cannot be calculated with diff_method=None")
+
+    def execute_and_compute_jacobian(self, tapes: QuantumScriptBatch) -> tuple[ResultBatch, tuple]:
+        raise qml.QuantumFunctionError("Derivatives cannot be calculated with diff_method=None")
+
+
```

Comment:
```suggestion
class NoGradients(JacobianProductCalculator):
    """A jacobian product calculator that raises errors when a vjp or jvp is requested."""
    
    error_msg = "Derivatives cannot be calculated with diff_method=None"

    def compute_jacobian(self, tapes: QuantumScriptBatch) -> tuple:
        raise qml.QuantumFunctionError(NoGradients.error_msg)

    def compute_vjp(self, tapes: QuantumScriptBatch, dy: Sequence[Sequence[TensorLike]]) -> tuple:
        raise qml.QuantumFunctionError(NoGradients.error_msg)

    def execute_and_compute_jvp(
        self, tapes: QuantumScriptBatch, tangents: Sequence[Sequence[TensorLike]]
    ) -> tuple[ResultBatch, tuple]:
        raise qml.QuantumFunctionError(NoGradients.error_msg)

    def execute_and_compute_jacobian(self, tapes: QuantumScriptBatch) -> tuple[ResultBatch, tuple]:
        raise qml.QuantumFunctionError(NoGradients.error_msg)


```
Not very important but since msg are the same we should manage them together instead of hardcode